cmd/sentry/download/downloader.go

```diff
@@ -470,30 +471,42 @@ func (cs *ControlServerImpl) blockHeaders(ctx context.Context, pkt eth.BlockHead
 			Number:    number,
 		})
 	}
-
 	if segments, penaltyKind, err := cs.Hd.SplitIntoSegments(csHeaders); err == nil {
 		if penaltyKind == headerdownload.NoPenalty {
-			var canRequestMore bool
-			for _, segment := range segments {
-				requestMore, penalties := cs.Hd.ProcessSegment(segment, false /* newBlock */, ConvertH256ToPeerID(peerID))
-				canRequestMore = canRequestMore || requestMore
-				if len(penalties) > 0 {
-					cs.Penalize(ctx, penalties)
+			if cs.Hd.GetBackwards() {
```

Comment:
I'm a bit confused about this backwards flag. Firstly, my understanding is that the existing (pre-Merge) downloader downloads headers mostly (besides skeleton requests) backwards, towards the genesis. So "backwards" meaning PoS sync is a bit misleading. Secondly, why is it all or nothing mode? Couldn't exclusively backwards downloading of PoS blocks run simultaneously with the old skeleton+backwards algorithm for pre-Merge blocks? To sum up, to my mind "GetBackwards" is a confusing name.

Reply:
I changed it to POSSync, the flag indicates when we are doing sync for proof-of-stake or not

Reply:
So does it mean that we can't download PoS and PoW headers simultaneously?

Reply:
we can, if we assume that the block supplied by the beacon chain is correct, we sync backwards until we find a block whose parenthash is present in our database, this can be past the terminal block. if such thing happen, we unwind

Reply:
Hmm. What about a node that wants to sync from scratch some time after the Merge? It doesn't have any PoW headers in the DB yet.

Reply:
It will first sync forward like we do already normally in POW, and after we surpass totalTerminalDifficulty, the node will stand still and wait for payloads and then proceed to backward sync once it receives a payload. a while after the merge we may even remove forward sync and have just downward sync.

Reply:
You can try this, if you follow the procedure i pointed out above in the PR.

Reply:
Well, the problem is that our current (PoW) downloader is not a forward one (and the name `HeadersForward` is incorrect). As far as I understand it, the current downloader first sends a skeleton request and then starts extending resulting anchors downwards, towards genesis. So it's not correct that we first download in the forward direction, then reach the Terminal Total Difficulty (TTD), and then switch to the PoS header downloader in the downwards direction. And after the merge the skeleton request will likely return some PoS blocks pass the TTD, won't it?

Reply:
Okay, i think that headersForward means that headers are inserted in "forward" order, if you were remove preverified hashes, you would have to consensus check in forward order(genesis to head), however in our case we process the headers in downward order we start from head and go to genesis

Reply:
If you think that `headersFoward` is wrong/misleading that can be refactored in another PR, it is not in the scope of this one i believe

Reply:
We need to insert them backwards to check their validity, we need to check if previous block parenthash is equal to current block hash